### PR TITLE
ls: Match the gnu behavior for colors

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

**Comment by sylvestre** [comment]
it is failing but I would like to get the results

File: Cargo.toml [hunk 1]
```diff
@@ -289,6 +289,7 @@ itertools = "0.12.0"
 libc = "0.2.150"
 lscolors = { version = "0.16.0", default-features = false, features = [
   "nu-ansi-term",
```
Comment (cakebaker) [inline Cargo.toml:291]: Is this still needed? At least the tests pass without it.

File: src/uu/ls/Cargo.toml [hunk 2]
```diff
@@ -22,7 +22,7 @@ number_prefix = { workspace = true }
 uutils_term_grid = { workspace = true }
 terminal_size = { workspace = true }
 glob = { workspace = true }
-lscolors = { workspace = true }
+lscolors = { workspace = true, features = ["gnu_legacy"] }
```
Comment (cakebaker) [inline src/uu/ls/Cargo.toml:25]: Is this necessary? We didn't have the `nu-ansi-term` feature mentioned here.

File: Cargo.toml (hunk 1 above)
Comment (sylvestre) [reply Cargo.toml:291]: probably not, thanks :)

**Review by cakebaker — COMMENTED** [review]
I'm not sure whether this is the intended behavior or a bug. When running `ls --color=always` and redirecting the output to a file, the output of uutils `ls` and GNU `ls` differs. All but the first escape code are different, with the ones from uutils `ls` being prefixed with `^[[0m`.

**Comment by sylvestre** [comment]
how do you compare the output ?
i did 
```
ls > out
cat -v out
```

**Comment by cakebaker** [comment]
Here the steps I did:
```
$ mkdir example example/a example/b
$ cargo run ls --color=always example/ > color.txt
$ ls --color=always example/ > colorgnu.txt
$ cat -v color.txt colorgnu.txt 
^[[0m^[[01;34ma^[[0m
^[[0m^[[01;34mb^[[0m
^[[0m^[[01;34ma^[[0m
^[[01;34mb^[[0m
```

**Comment by sylvestre** [comment]
The difference is that GNU ls does not immediately reset the text style after each directory name. It only resets (^[[0m) before starting a new colored text, or at the end of the output.

File: tests/by-util/test_ls.rs [hunk 3]
```diff
@@ -2020,12 +2028,8 @@ fn test_ls_color() {
...
 
     // Nested dir should be shown and colored
-    scene
-        .ucmd()
-        .arg("--color")
-        .arg("a")
-        .succeeds()
-        .stdout_contains(nested_dir_with_colors);
+    let result = scene.ucmd().arg("--color").arg("a").succeeds();
+    assert!(result.stdout_str().contains(nested_dir_with_colors));
```
[the remark this answers is not in the document]
Comment (sylvestre) [reply tests/by-util/test_ls.rs:2032]: same, easier to debug

File: tests/by-util/test_ls.rs [hunk 4]
```diff
@@ -2037,13 +2041,18 @@ fn test_ls_color() {
...
         .arg("-w=15")
         .arg("-C")
-        .succeeds()
-        .stdout_only(format!("{a_with_colors}  test-color\nb  {z_with_colors}\n"));
+        .succeeds();
+    let expected = format!("{}  test-color\x0ab  {}\n", a_with_colors, z_with_colors);
+    assert_eq!(
+        to_ascii_representation(result.stdout_str()),
+        to_ascii_representation(&expected)
+    );
```
[the remark this answers is not in the document]
Comment (sylvestre) [reply tests/by-util/test_ls.rs:2054]: yeah, it is also for debugging :)

File: tests/by-util/test_ls.rs [hunk 5]
```diff
@@ -2006,12 +2017,9 @@ fn test_ls_color() {
...
     for param in ["--color", "--col", "--color=always", "--col=always"] {
-        scene
-            .ucmd()
-            .arg(param)
-            .succeeds()
-            .stdout_contains(a_with_colors)
-            .stdout_contains(z_with_colors);
+        let result = scene.ucmd().arg(param).succeeds();
+        assert!(result.stdout_str().contains(a_with_colors));
+        assert!(result.stdout_str().contains(z_with_colors));
```
[the remark this answers is not in the document]
Comment (tertsdiepraam) [reply tests/by-util/test_ls.rs:2022]: Sounds like an issue with our test functions then 😄 Is there some info missing in the old version?

File: tests/by-util/test_ls.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (sylvestre) [reply tests/by-util/test_ls.rs:2054]: sure, i didn't know about escape_default :)

File: tests/by-util/test_ls.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (sylvestre) [reply tests/by-util/test_ls.rs:2022]: no, it is just that it is easier when you want to do some analysis on the stdout / stderr

**Review by tertsdiepraam — APPROVED** [review]
Nice! The spell-check is failing but once that's fixed LGTM!

**Pull request merged**